backend/src/main/java/botobo/core/quiz/application/QuizService.java

```diff
@@ -27,26 +24,33 @@ public QuizService(WorkbookRepository workbookRepository, CardRepository cardRep
         this.cardRepository = cardRepository;
     }
 
-    public List<QuizResponse> createQuiz(List<Long> ids) {
-        List<Card> cards = new ArrayList<>();
-        for (Long id : ids) {
-            final Workbook workbook = workbookRepository.findById(id)
-                    .orElseThrow(WorkbookNotFoundException::new);
-            cards.addAll(workbook.getCardsAsList());
+    public List<QuizResponse> createQuiz(List<Long> workbookIds) {
+        validateWorkbookIds(workbookIds);
+
+        final List<Card> nextShownCards = cardRepository.findNextCardAndWorkbookIdIn(workbookIds);
+        Cards nextCards = new Cards(nextShownCards);
+        if (nextCards.size() == 10) {
+            nextShownCards.forEach(Card::cancelNextQuiz);
+            return QuizResponse.cardsOf(nextCards);
         }
-        final int maxLimit = Math.min(cards.size(), 10);
-        Collections.shuffle(cards);
-        final List<Card> quiz = cards.stream().limit(maxLimit).collect(Collectors.toList());
-        return QuizResponse.listOf(quiz);
+
+        Cards cards = new Cards(cardRepository.findCardsByWorkbookId(workbookIds)).shuffle();
+        final int index = Math.min(cards.size(), 10 - nextCards.size());
+
+        Cards quiz = new Cards(cards.subList(index));
+        quiz.addAll(nextCards);
+        nextShownCards.forEach(Card::cancelNextQuiz);
+        return QuizResponse.cardsOf(quiz);
```

Comment:
```suggestion
       Cards cards = new Cards(cardRepository.findCardsByWorkbookId(workbookIds));
        return QuizResponse.listOf(cards.getCardsForQuiz());
```

저는 이렇게 생각했습니다!
findCardsByWorkbookId(workbookIds)로 workbook id 리스트에 해당되는 모든 card를 가져옵니다.
-> shuffle을 합니다.
-> 그 카드 중 isNextQuiz가 true인 것을 10개를 뽑습니다.
-> 만약 10개가 충족되지 않았을 경우 나머지 카드 중 isNextQuiz가 false인 카드를 총 10개가 될 때까지 뽑습니다.
-> 이걸 반환해서 QuizResponse를 만듭니다. 

조앤이 구현해준 것과 다른점은 쿼리를 한 번 날리는 것? 
그리고 가져온 카드들을 이용해 퀴즈를 만드는 책임을 Cards 일급컬렉션에 위임할 수 있다는 것?
정도가 될 수 있을 것 같습니다!!
다만 이렇게 되면 isNextQuiz가 true인 카드가 10개가 된다해도 무조건 다 들고 온다는 단점이 있겠네요ㅠ

Reply:
좋은 의견 감사해요!! 다른분들 의견도 다 확인하구 절충해서 새로 커밋&푸시해볼게요!! 감사합니당 ㅎㅎ